packages/components/src/components/Button/Button.tsx

```diff
@@ -0,0 +1,103 @@
+
+import { buttonIconSizeMap, buttonStyles } from './styles/buttonStyles';
+import { iconStyles } from './styles/iconStyles';
+import { textStyles } from './styles/textStyles';
+
+type ButtonRounding = 'medium' | 'full';
+
+type ButtonStyle = 'default' | 'positive' | 'negative' | 'danger' | 'loud';
+
+type ButtonSize = 'small' | 'medium' | 'large';
+
+type ButtonType = 'button' | 'submit' | 'reset';
+
+type ButtonEmphasis = 'default' | 'weak' | 'strong';
+
+export type ButtonProps = {
+    type?: ButtonType;
+    title?: string;
+    style?: ButtonStyle;
+    emphasis?: ButtonEmphasis;
+    hideLabel?: boolean;
+    size?: ButtonSize;
+    rounding?: ButtonRounding;
+    disabled?: boolean;
+    icon?: ReactElement;
+    children?: ReactNode;
+    onClick?: (event?: MouseEvent<HTMLButtonElement>) => void;
+    hugWidth?: boolean;
+    'aria-label'?: string;
+    'aria-describedby'?: string;
+    formId?: string;
+    'data-test-id'?: string;
+};
+
+export const ButtonComponent = (
+    {
+        type = 'button',
+        style = 'default',
+        size = 'medium',
+        rounding = 'medium',
+        emphasis = 'strong',
+        title = '',
+        hideLabel = false,
+        disabled = false,
+        icon,
+        children,
+        onClick,
+        hugWidth = true,
+        formId,
+        'aria-label': ariaLabel,
+        'aria-describedby': ariaDescribedBy,
+        'data-test-id': dataTestId = 'fondue-button',
+    }: ButtonProps,
+    ref: ForwardedRef<HTMLButtonElement | null>,
+) => {
+    return (
+        <button
+            data-test-id={dataTestId}
+            aria-label={ariaLabel}
+            aria-disabled={disabled}
+            aria-describedby={ariaDescribedBy}
+            ref={ref}
+            form={formId}
+            title={title}
+            type={type}
+            onClick={onClick}
+            disabled={disabled}
+            className={buttonStyles({
+                disabled,
+                rounding,
+                size,
+                hugWidth,
+                emphasis,
+                style,
+                iconOnly: (icon && !children) || hideLabel,
+            })}
+        >
+            {icon && (
+                <span
+                    data-test-id={`${dataTestId}-icon`}
+                    className={iconStyles({
+                        iconSpacing: children && !hideLabel ? size : 'none',
+                        emphasis,
+                        style,
+                    })}
+                >
+                    {cloneElement(icon, { size: buttonIconSizeMap[size] })}
+                </span>
+            )}
+            {children && (
+                <span className={textStyles({ hideLabel, emphasis, style })} data-test-id={`${dataTestId}-text`}>
+                    {children}
+                </span>
+            )}
+        </button>
+    );
+};
+
+export const Button = forwardRef<HTMLButtonElement | null, ButtonProps>(ButtonComponent);
```

Comment:
What about removing the extra `const` and extra typing `ref: ForwardedRef<HTMLButtonElement | null>` and the props `}: ButtonProps,`? Like this

<img width="690" alt="Screenshot 2024-04-15 at 10 39 19" src="https://github.com/Frontify/fondue/assets/6846950/df0a75a8-dcdd-4e17-a82f-aca965c78aae">

Reply:
This will not allow us to access the inner component and it's type directly and break the docs generation in storybook. 

Apart from that i like to have this separated for improved readability

Reply:
It doesn't break actually, and for the readability, I would let the formatter do it and probably you could use the spread operator a bit more instead of destructuring everything.

I did the exercise (spreading, adding `defaultVariants` in the `sv`), and instead of the 103 lines, we have 68 (a bit more readable) lines.

![Screenshot 2024-04-15 at 13 31 37](https://github.com/Frontify/fondue/assets/6846950/b5c722d0-25dc-441f-8a1b-ab7b9045f19a)